aesara/tensor/math_opt.py

```diff
@@ -226,14 +226,19 @@ def local_func_inv(fgraph, node):
         if is_inverse_pair(node_op, prev_op, inv_pair):
             # We don't need to copy stack trace, because the optimization
             # is trivial and maintains the earlier stack trace
-            return x.owner.inputs
+            ottype = node.out.dtype
+            inp = x.owner.inputs[0]
+            # Functions may have casted integer input to float
+            if inp.dtype != ottype:
+                inp = cast(inp, ottype)
+            return [inp]
 
     return
 
 
 @register_canonicalize
 @register_specialize
-@local_optimizer([Elemwise])
+@local_optimizer([log, exp])
```

Comment:
@brandonwillard Should I revert this until #557 is sorted?

Reply:
Yeah, because when `tracks` is used, this will likely limit the optimization to only these exact instances.

Reply:
Just to be clear, what would other instances be? `inplace` versions or something else?

Reply:
The reason I am asking is because I feel the `isinstance` checks inside would be no less restrictive that these `tracks`.